docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Conversion/MapKeyToValueInputOutputPair.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using Microsoft.ML.SamplesUtils;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public class MapKeyToValueInputOutputPair
+    {
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Create a list of data examples.
+            var examples = DatasetUtils.GenerateRandomMulticlassClassificationExamples(1000);
+
+            // Convert the examples list to an IDataView object, which is consumable by ML.NET API.
+            var dataView = mlContext.Data.LoadFromEnumerable(examples);
+
+            //////////////////// Data Preview ////////////////////
+            // Label    Features
+            // AA       0.7262433,0.8173254,0.7680227,0.5581612,0.2060332,0.5588848,0.9060271,0.4421779,0.9775497,0.2737045
+            // BB       0.4919063,0.6673147,0.8326591,0.6695119,1.182151,0.230367,1.06237,1.195347,0.8771811,0.5145918
+            // CC       1.216908,1.248052,1.391902,0.4326252,1.099942,0.9262842,1.334019,1.08762,0.9468155,0.4811099
+            // DD       0.7871246,1.053327,0.8971719,1.588544,1.242697,1.362964,0.6303943,0.9810045,0.9431419,1.557455
+
+            // Create a pipeline. 
+            var pipeline =
+                    // Convert the string labels into key types.
+                    mlContext.Transforms.Conversion.MapValueToKey("Label")
+                    // Apply StochasticDualCoordinateAscent multiclass trainer.
+                    .Append(mlContext.MulticlassClassification.Trainers.SdcaMaximumEntropy());
+
+            // Split the data into training and test sets. Only training set is used in fitting
+            // the created pipeline. Metrics are computed on the test.
+            var split = mlContext.Data.TrainTestSplit(dataView, testFraction: 0.1);
+
+            // Train the model and do predictions on the test set.
+            var dataWithPredictions = pipeline.Fit(split.TrainSet).Transform(split.TestSet);
+
+            // at this point, the Label colum is tranformed from strings, to DataViewKeyType and
+            // the transformation has added the PredictedLabel column, with 
+            var newPipeline = mlContext.Transforms.Conversion.MapKeyToValue(new[] 
```

Comment:
>newPipeline [](start = 16, length = 11)

i am confused by this -- the `newPipeline`  has  no interaction with the previous `pipeline`.  

So how does the `newPipeline` know about the mapping  that  `pipeline` generated in `MapValueToKey`

Reply:
That's what MapKeyToValue does. The mapping is saved in the Annotations of the column. 

---
In reply to: [271831815](https://github.com/dotnet/machinelearning/pull/3187#discussion_r271831815) [](ancestors = 271831815)

Reply:
ah i see.. makes sense..

we may want to add that as comment in the sample .. am sure users will have the same question 

---
In reply to: [271866086](https://github.com/dotnet/machinelearning/pull/3187#discussion_r271866086) [](ancestors = 271866086,271831815)